docs/guide/laravel.md

```diff
@@ -204,6 +204,145 @@ Content-Type: application/json
 Controller-Method: ArticleController@getList
 Model-Method: Article::getList
 ```
+  
+## Performance Tuning SQL Queries
+
+SQL tuning is the process of improving SQL queries to accelerate your servers performance.
+It's general purpose is to reduce the amount of time it takes a user to 
+receive a result after issuing a query, and to reduce the amount of 
+resources used to process a query. 
+
+Big SQL queries can suffer performance issues. 
+Performance degradation often occurs if the database is not being properly 
+maintained or if queries can be rewritten more efficiently.
+
+Following Query optimization tips for better performance
+
+- **Apply Index On Necessary Columns:**
+
+Table indexes in databases help retrieve information faster and more efficiently.
+An index creates a unique data column without overlapping each other. 
+It improves the speed of data retrieval.
+
+- **Avoid Query In A Loop:**
+ 
+ If sometimes we need to execute the same query multiple times like if 
+ we need to insert 10 records in any table at that time don’t use 
+ insert query in the loop instead of using bulk insert.
+ 
+ We shouldn't call Eloquent relationships method in loop. 
+ First convert model object in to collection and then pass in to loop.
+ 
+  
+ **Example:**
+ ```php
+ 
+ 
+ public function category()
+     {
+         return $this->belongsTo(Category::class,
+             'catgeory_id', 'id'
+         );
+     }
+     
+ public function getList()
+     {
+  
+     $products = Product::with(['catgory'])->all();
+     
+     $collect_products = collect($products);
+     
+     foreach ($collect_products as $key => $product)
+        {
+     
+             if($product && $product->category){
+             
+             }
+         
+        }
+        
+     } 
+
+ ```
+ 
+
+- **Avoid too many JOINs:**
+
+When you add multiple tables to a query and join them, you may overload it. 
+In addition, a large number of tables to retrieve data from may result in an 
+inefficient execution plan. When generating a plan, the SQL query optimizer 
+needs to identify how the tables are joined, in which order, how and when to 
+apply filters and aggregation.
+
+JOIN elimination is one of the many techniques to achieve efficient query plans. 
+You can split a single query into several separate queries which can later be joined, 
+and thus remove unnecessary joins, subqueries, tables, etc.
+
+- **Use SELECT fields instead of retrieve all data:**
+
+The SELECT statement is used to retrieve data from the database. 
+In the case of large databases, it is not recommended to retrieve all data 
+because this will take more resources on querying a huge volume of data.
+
+If we execute the following query, we will retrieve all data from the Users table, 
+including, for example, users’ avatar pictures. The result table will contain 
+lots of data and will take too much memory and CPU usage.
+
```

Comment:
@we-vikram-wri231 add a heading as `**BAD CODE**`

Reply:
@themodernpk Sir,
Added **BAD CODE**